File: actividad19.java
import java.util.Scanner;

public class LinkedList {

    class Node{
        String data;
        Node next;
    }

    private Node root;

    LinkedList(){
        root = null;
    }

    private void Ingresar(String item){
        Node nuevo = new Node();
        nuevo.data = item;

        //SI LA LISTA ESTA VACIA...
        if(root == null){
            root = nuevo;
        }else{
            Node pointer = root;
            while (pointer.next != null){
                pointer = pointer.next;
            }
            pointer.next = nuevo;
        }
    }

    private void pop(String item){

        //CASO 1 - DESTRUIR LA RAIZ
        if(root.data == item){
            root = root.next;
        }

        //CASO 2 - DESTRUIR EL DATO PEDIDO
        else{
            Node pointer = root;
            while (pointer != null){
                if(pointer.next.data == item){
                    pointer = pointer.next.next;
                }
                pointer = pointer.next;
            }
        }
    }

    //CASO 3 - DESTRUIR EL ULTIMO NODO
    private void pop(){


    }

    private void peek(String item){
        Node pointer = root;
        boolean found = false;
        while (pointer != null){
            if(pointer.data == item){
                found = true;
            }
            pointer = pointer.next;
        }
        if (found)
            System.out.println("Elemento encontrado");
        else
            System.out.println("Elemento no encontrado");

    }

    private void Imprimir(){
        Node pointer = root;
        String salida = "";
        while (pointer != null){
            salida = salida + pointer.data + " - ";
            pointer = pointer.next;
        }
        System.out.println(salida);
    }

    public static void main(String[] args){
        Scanner input = new Scanner(System.in);
        LinkedList ll = new LinkedList();
        ll.Ingresar("uno");
        ll.Ingresar("dos");
        ll.Ingresar("tres");
        ll.Ingresar("cuatro");
        ll.Imprimir();
        ll.Ingresar("cinco");
        ll.Ingresar("seis");
        ll.peek("dos");
        ll.peek("uno");
        ll.peek("cinco");
        ll.Imprimir();
        ll.pop("uno");
        ll.Imprimir();
        ll.pop("cuatro");
        ll.Imprimir();
    }
}

File: actividad19.py
#Listas Enlazadas

class Node:
    def __init__(self):
        self.data = None
        self.next = None

class LinkedList:
    def __init__(self):
        self.root = None
        self.q = None

    def push(self,item):
        nodo = Node()
        nodo.data = item
        if self.root is None:
            self.root = nodo
            self.q = nodo
        else:
            pointer = self.root
            while pointer.next is not None:
                pointer = pointer.next
            pointer.next = nodo
            self.q = pointer.next

    def peek(self):
        pointer = self.root
        while pointer.next is not None:
            print (pointer.data)
            pointer = pointer.next
        print (pointer.data)

    def pop(self,item):
        if self.root is None:
            print("Lista vacia")
        else:
            _continue = True
            pointer = self.root
            while _continue:

                #CASO 1 - CUANDO SE ELIMINA LA RAIZ
                if pointer == self.root and pointer.data == item:
                    del self.root
                    self.root = pointer.next
                    break
                else:
                    #CASO 2 - CUANDO SE ELIMINA CUALQUIER NODO
                    if pointer.next.next is not None and pointer.next.data == item:
                        r = pointer.next
                        pointer.next = r.next
                        del r
                        break
                    else:
                        #CASO 3 - CUANDO SE ELIMINA EL ULTIMO NODO
                        if pointer.next == self.q and self.q.data == item:
                            del self.q
                            self.q = None
                            pointer.next = None
                            break
                        else:
                            pointer = pointer.next

                    
#Programa
ll = LinkedList()
def menu():
    print("1) Push\n2) Peek\n3) Pop\n4) Exit")
    enter = int(input("-> "))
    if enter == 1:
        item = input("Ingrese el valor: ")
        ll.push(item)
        menu()
    elif enter == 2:
        ll.peek()
        menu()
    elif enter == 3:
        item = input("Ingrese el valor: ")
        ll.pop(item)
        menu()
    else:
        print("exit...")
menu()
